proxy/proxy.go

```diff
@@ -1184,7 +1184,12 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 				Duration:     time.Since(ctx.startServe),
 			}
 
-			logging.LogAccess(entry)
+			extraData := ctx.stateBag[al.AccessLogMetadataKey]
+			if extraData == nil {
+				extraData = make(map[string]string)
+			}
+
+			logging.LogAccess(entry, extraData.(map[string]string))
```

Comment:
I would rather do:
```
extraData := ctx.stateBag[al.AccessLogMetadataKey]  // can be nil is fine
logging.LogAccess(entry, extraData)
```

Reply:
If we do this, we're saying type of `extraData` is `interface{}`. I don't really like that idea...

Reply:
logrus.Fields is also map[string]interface{}, so why converting everything to map[string]string and later pass it into map[string]interface{}?
This will either be changed in the compiler, if it's smart enough or slow down the performance in the critical path.

Reply:
I've changed it to map[string]interface{}. I just didn't want to use `logrus.Fields`. But if you think that's a better approach, I'm happy to do it.